core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -531,6 +510,49 @@ class TeamPolicyInternal<Kokkos::Cuda, Properties...>
     return *this;
   }
 #endif
+
+  template <class ClosureType, class FunctorType, class BlockSizeCallable>
+  int internal_team_size_common(const FunctorType& f,
+                                BlockSizeCallable&& block_size_callable) const {
+    using closure_type = ClosureType;
+    using functor_value_traits =
+        Impl::FunctorValueTraits<FunctorType, typename traits::work_tag>;
+
+    cudaFuncAttributes attr =
+        CudaParallelLaunch<closure_type, typename traits::launch_bounds>::
+            get_cuda_func_attributes();
+    const int block_size =
+        block_size_callable(space().impl_internal_space_instance(), attr, f,
+                            (size_t)vector_length(),
+                            (size_t)team_scratch_size(0) + 2 * sizeof(double),
+                            (size_t)thread_scratch_size(0) + sizeof(double) +
+                                ((functor_value_traits::StaticValueSize != 0)
+                                     ? 0
+                                     : functor_value_traits::value_size(f)));
+    assert(block_size > 0);
```

Comment:
`KOKKOS_ASSERT` macro should probably be preferred over the standard one, I think? Not a blocker; we can fix this later if necessary